README.md
# anitabakai


gallery2/changelog.txt
GalleryJack Rebooted

Version 1.6

23-Sep-2013

// Updated to jQuery 1.10.2


****************************************************************

Version 1.5

4-Aug-2013

// Updated to jQuery 1.10.1
// Fixed z-index issue with local video player + fixed-sized mode


****************************************************************

Version 1.4

12-Apr-2013

// Updated to jQuery 1.9.1


****************************************************************

Version 1.3

12-Jan-2013

// Updated to jQuery 1.8.3


****************************************************************

Version 1.2

15-Mar-2012

// Corrected z-index bug for fixed-size version


****************************************************************


GalleryJack Rebooted

Version 1.1

12-Dec-2011

// Updated to jQuery 1.7.1

// Fixed IE8 Memory Leak

// Fixed bug when no thumbnails are used

js/custom.js
/* ### Ajax images ##################################################################### */
jQuery( document ).ready(function() {
	// browser window scroll (in pixels) after which the "back to top" link is shown
	var offset = 300,
		//duration of the top scrolling animation (in ms)
		scroll_top_duration = 700,
		//grab the "back to top" link
		jQueryback_to_top = jQuery('.back-to-top');

	//hide or show the "back to top" link
	jQuery(window).scroll(function(){
		( jQuery(this).scrollTop() > offset ) ? jQueryback_to_top.addClass('cd-is-visible') : jQueryback_to_top.removeClass('cd-is-visible cd-fade-out');
	});

	//smooth scroll to top
	jQueryback_to_top.on('click', function(event){
		event.preventDefault();
		jQuery('body,html').animate({
			scrollTop: 0 ,
		 	}, scroll_top_duration
		);
	});
});

jQuery(document).ready(function(jQuery){
	//open interest point description
	jQuery('.AnitaBakai-single-point').children('a').on('click', function(){
		var selectedPoint = jQuery(this).parent('li');
		if( selectedPoint.hasClass('is-open') ) {
			selectedPoint.removeClass('is-open').addClass('visited');
		} else {
			selectedPoint.addClass('is-open').siblings('.AnitaBakai-single-point').removeClass('is-open');
		}
	});
	//close interest point description
	jQuery('.AnitaBakai-close-info').on('click', function(event){
		event.preventDefault();
		jQuery(this).parents('.AnitaBakai-single-point').eq(0).removeClass('is-open').addClass('visited');
	});
});

/* ### Ajax images ##################################################################### */
var elements = document.getElementsByClassName('async-image');
Array.prototype.filter.call(elements, function(element){
    var img = new Image(); 
    img.onload = function() { 
       element.setAttribute("src",img.src);
       var title = document.getElementById('title');
       var title2 = document.getElementById('title2');
       var overlay = document.getElementById('overlay');
       if ( title ) {
       		//title.removeAttribute('style');
       		title.className = "title";
       }         
       if ( title2 ) {
       		title2.className = "title";
       }       
       if ( overlay ) {
       		//overlay.removeAttribute('style');
       		overlay.className = "overlay";
       };       
    }
    var new_src = element.getAttribute("data-src");
    img.src = new_src;
}); 


/* ### Fancy Search Form ##################################################################### */
jQuery( document ).ready(function( $ ) {
	new UISearch( document.getElementById( 'AnitaBakai-search' ) );
	jQuery('#st-container').load('nav.html', function() {
		//Begin: MailChimp JS
		jQuery('#invite').ketchup().submit(function(evt) {
			evt.preventDefault();
			if (jQuery(this).ketchup('isValid')) {
				var action = jQuery(this).attr('action');

				jQuery.ajax({
					url: action,
					type: 'POST',
					data: {
						email: jQuery('#address').val()
					},
					success: function(data){
						jQuery('#result').html(data).css('color', '#35cf76');
					},
					error: function() {
						jQuery('#result').html('Sorry, an error occurred.').css('color', '#e74c3c');
					}
				});
			}else{
				jQuery('#result').html('Please enter an valid email address.').css('color', '#e74c3c');
			}
			return false;
		});
		//End: MailChimp JS
	});
	jQuery( "#trigger-menu" ).click(function() {
		jQuery('.AnitaBakai-search').fadeToggle('fast');
	});
});

// http://stackoverflow.com/a/11381730/989439
function mobilecheck() {
	var check = false;
	(function(a){if(/(android|ipad|playbook|silk|bb\d+|meego).+mobile|avantgo|bada\/|blackberry|blazer|compal|elaine|fennec|hiptop|iemobile|ip(hone|od)|iris|kindle|lge |maemo|midp|mmp|netfront|opera m(ob|in)i|palm( os)?|phone|p(ixi|re)\/|plucker|pocket|psp|series(4|6)0|symbian|treo|up\.(browser|link)|vodafone|wap|windows (ce|phone)|xda|xiino/i.test(a)||/1207|6310|6590|3gso|4thp|50[1-6]i|770s|802s|a wa|abac|ac(er|oo|s\-)|ai(ko|rn)|al(av|ca|co)|amoi|an(ex|ny|yw)|aptu|ar(ch|go)|as(te|us)|attw|au(di|\-m|r |s )|avan|be(ck|ll|nq)|bi(lb|rd)|bl(ac|az)|br(e|v)w|bumb|bw\-(n|u)|c55\/|capi|ccwa|cdm\-|cell|chtm|cldc|cmd\-|co(mp|nd)|craw|da(it|ll|ng)|dbte|dc\-s|devi|dica|dmob|do(c|p)o|ds(12|\-d)|el(49|ai)|em(l2|ul)|er(ic|k0)|esl8|ez([4-7]0|os|wa|ze)|fetc|fly(\-|_)|g1 u|g560|gene|gf\-5|g\-mo|go(\.w|od)|gr(ad|un)|haie|hcit|hd\-(m|p|t)|hei\-|hi(pt|ta)|hp( i|ip)|hs\-c|ht(c(\-| |_|a|g|p|s|t)|tp)|hu(aw|tc)|i\-(20|go|ma)|i230|iac( |\-|\/)|ibro|idea|ig01|ikom|im1k|inno|ipaq|iris|ja(t|v)a|jbro|jemu|jigs|kddi|keji|kgt( |\/)|klon|kpt |kwc\-|kyo(c|k)|le(no|xi)|lg( g|\/(k|l|u)|50|54|\-[a-w])|libw|lynx|m1\-w|m3ga|m50\/|ma(te|ui|xo)|mc(01|21|ca)|m\-cr|me(rc|ri)|mi(o8|oa|ts)|mmef|mo(01|02|bi|de|do|t(\-| |o|v)|zz)|mt(50|p1|v )|mwbp|mywa|n10[0-2]|n20[2-3]|n30(0|2)|n50(0|2|5)|n7(0(0|1)|10)|ne((c|m)\-|on|tf|wf|wg|wt)|nok(6|i)|nzph|o2im|op(ti|wv)|oran|owg1|p800|pan(a|d|t)|pdxg|pg(13|\-([1-8]|c))|phil|pire|pl(ay|uc)|pn\-2|po(ck|rt|se)|prox|psio|pt\-g|qa\-a|qc(07|12|21|32|60|\-[2-7]|i\-)|qtek|r380|r600|raks|rim9|ro(ve|zo)|s55\/|sa(ge|ma|mm|ms|ny|va)|sc(01|h\-|oo|p\-)|sdk\/|se(c(\-|0|1)|47|mc|nd|ri)|sgh\-|shar|sie(\-|m)|sk\-0|sl(45|id)|sm(al|ar|b3|it|t5)|so(ft|ny)|sp(01|h\-|v\-|v )|sy(01|mb)|t2(18|50)|t6(00|10|18)|ta(gt|lk)|tcl\-|tdg\-|tel(i|m)|tim\-|t\-mo|to(pl|sh)|ts(70|m\-|m3|m5)|tx\-9|up(\.b|g1|si)|utst|v400|v750|veri|vi(rg|te)|vk(40|5[0-3]|\-v)|vm40|voda|vulc|vx(52|53|60|61|70|80|81|83|85|98)|w3c(\-| )|webc|whit|wi(g |nc|nw)|wmlb|wonu|x700|yas\-|your|zeto|zte\-/i.test(a.substr(0,4)))check = true})(navigator.userAgent||navigator.vendor||window.opera);
	return check;
}

// uisearch.js v1.0.0
;( function( window ) {
	
	'use strict';
	
	!window.addEventListener && window.Element && (function () {
	   function addToPrototype(name, method) {
		  Window.prototype[name] = HTMLDocument.prototype[name] = Element.prototype[name] = method;
	   }
	 
	   var registry = [];
	 
	   addToPrototype("addEventListener", function (type, listener) {
		  var target = this;
	 
		  registry.unshift({
			 __listener: function (event) {
				event.currentTarget = target;
				event.pageX = event.clientX + document.documentElement.scrollLeft;
				event.pageY = event.clientY + document.documentElement.scrollTop;
				event.preventDefault = function () { event.returnValue = false };
				event.relatedTarget = event.fromElement || null;
				event.stopPropagation = function () { event.cancelBubble = true };
				event.relatedTarget = event.fromElement || null;
				event.target = event.srcElement || target;
				event.timeStamp = +new Date;
	 
				listener.call(target, event);
			 },
			 listener: listener,
			 target: target,
			 type: type
		  });
	 
		  this.attachEvent("on" + type, registry[0].__listener);
	   });
	 
	   addToPrototype("removeEventListener", function (type, listener) {
		  for (var index = 0, length = registry.length; index < length; ++index) {
			 if (registry[index].target == this && registry[index].type == type && registry[index].listener == listener) {
				return this.detachEvent("on" + type, registry.splice(index, 1)[0].__listener);
			 }
		  }
	   });
	 
	   addToPrototype("dispatchEvent", function (eventObject) {
		  try {
			 return this.fireEvent("on" + eventObject.type, eventObject);
		  } catch (error) {
			 for (var index = 0, length = registry.length; index < length; ++index) {
				if (registry[index].target == this && registry[index].type == eventObject.type) {
				   registry[index].call(this, eventObject);
				}
			 }
		  }
	   });
	})();
	
	!String.prototype.trim && (String.prototype.trim = function() {
		return this.replace(/^\s+|\s+$/g, '');
	});

	function UISearch( el, options ) {	
		this.el = el;
		this.inputEl = el.querySelector( 'form > input.AnitaBakai-search-input' );
		this._initEvents();
	}

	UISearch.prototype = {
		_initEvents : function() {
			var self = this,
				initSearchFn = function( ev ) {
					ev.stopPropagation();
					// trim its value
					self.inputEl.value = self.inputEl.value.trim();
					
					if( !classie.has( self.el, 'AnitaBakai-search-open' ) ) { // open it
						ev.preventDefault();
						self.open();
					}
					else if( classie.has( self.el, 'AnitaBakai-search-open' ) && /^\s*$/.test( self.inputEl.value ) ) { // close it
						ev.preventDefault();
						self.close();
					}
				}

			this.el.addEventListener( 'click', initSearchFn );
			this.el.addEventListener( 'touchstart', initSearchFn );
			this.inputEl.addEventListener( 'click', function( ev ) { ev.stopPropagation(); });
			this.inputEl.addEventListener( 'touchstart', function( ev ) { ev.stopPropagation(); } );
		},
		open : function() {
			var self = this;
			classie.add( this.el, 'AnitaBakai-search-open' );
			// focus the input
			if( !mobilecheck() ) {
				this.inputEl.focus();
			}
			// close the search input if body is clicked
			var bodyFn = function( ev ) {
				self.close();
				this.removeEventListener( 'click', bodyFn );
				this.removeEventListener( 'touchstart', bodyFn );
			};
			document.addEventListener( 'click', bodyFn );
			document.addEventListener( 'touchstart', bodyFn );
		},
		close : function() {
			this.inputEl.blur();
			classie.remove( this.el, 'AnitaBakai-search-open' );
		}
	}

	window.UISearch = UISearch;
} )( window );

function resizeImages(){
    //FIGURE IMAGE
    var figure = $("figure");

    //console.log(jQuery('.figure .img'));
    figure.each(function( index ) {
        figure = $(this);
        figureW = figure.width();
        figureH = figure.height();

        img = $('img', figure);
        imgW = img.width();
        imgH = img.height();

        img.css({
            top: '50%',
            left: '50%',
            'min-width': (figureW+80)+'px',
            margin: '-'+imgH/2+'px 0 0 -'+((imgW/2))+'px'
        });

        /*if(imgW < figureW){
         img.css({
         width: '100%',
         height: 'auto'
         });
         }else if(imgH < figureH){
         img.css({
         width: 'auto',
         height: '100%'
         });
         }
        console.log('('+figureW+', '+imgW+')' + '('+figureH+', '+imgH+')');*/
    });

    var wW = $( window ).width();
    var wH = $( window ).height();

    $("#slider1_container").css({
        width: wW+'px',
        height: wH+'px',
        border: '10px solid red'
    });
}

$( document ).ready(function() {
    resizeImages();
});

(function($){
    $(window).load(function(){
       $(".st-menu").niceScroll();
    });
})(jQuery);

$(document).ready(

    function() {

        $(".st-menu").niceScroll();

    }

);

$( window ).resize(function() {
    //resizeImages();
});




js/contact.js
// ~~~~~~~ Async images ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ //
var elements = document.getElementsByClassName('async-image');
Array.prototype.filter.call(elements, function(element){
    var img = new Image(); 
    img.onload = function() { 
       element.setAttribute("src",img.src);
       var title = document.getElementById('title');
       var overlay = document.getElementById('overlay');

       if ( title ) {
       		title.removeAttribute('style');
       };       
       if ( overlay ) {
       		overlay.removeAttribute('style');
       };
    }
    var new_src = element.getAttribute("data-src");
    img.src = new_src;
});


jQuery(window).load(function() { // makes sure the whole site is loaded
	jQuery('.index-contact .title').css("z-index","201");
});

jQuery('.async-image').load(function() {
    jQuery('.title').fadeIn();
});

// ~~~~~~~ Fancy Search Form ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ //
jQuery( document ).ready(function( $ ) {
	new UISearch( document.getElementById( 'AnitaBakai-search' ) );
	jQuery('#st-container').load('nav.html', function() {
		//Begin: MailChimp JS
		jQuery('#invite').ketchup().submit(function(evt) {
			evt.preventDefault();
			if (jQuery(this).ketchup('isValid')) {
				var action = jQuery(this).attr('action');

				jQuery.ajax({
					url: action,
					type: 'POST',
					data: {
						email: jQuery('#address').val()
					},
					success: function(data){
						jQuery('#result').html(data).css('color', '#35cf76');
					},
					error: function() {
						jQuery('#result').html('Sorry, an error occurred.').css('color', '#e74c3c');
					}
				});
			}else{
				jQuery('#result').html('Please enter an valid email address.').css('color', '#e74c3c');
			}
			return false;
		});
		//End: MailChimp JS
	});
	jQuery( "#trigger-menu" ).click(function() {
		jQuery('#AnitaBakai-search').fadeToggle('fast');
	});
});

// http://stackoverflow.com/a/11381730/989439
function mobilecheck() {
	var check = false;
	(function(a){if(/(android|ipad|playbook|silk|bb\d+|meego).+mobile|avantgo|bada\/|blackberry|blazer|compal|elaine|fennec|hiptop|iemobile|ip(hone|od)|iris|kindle|lge |maemo|midp|mmp|netfront|opera m(ob|in)i|palm( os)?|phone|p(ixi|re)\/|plucker|pocket|psp|series(4|6)0|symbian|treo|up\.(browser|link)|vodafone|wap|windows (ce|phone)|xda|xiino/i.test(a)||/1207|6310|6590|3gso|4thp|50[1-6]i|770s|802s|a wa|abac|ac(er|oo|s\-)|ai(ko|rn)|al(av|ca|co)|amoi|an(ex|ny|yw)|aptu|ar(ch|go)|as(te|us)|attw|au(di|\-m|r |s )|avan|be(ck|ll|nq)|bi(lb|rd)|bl(ac|az)|br(e|v)w|bumb|bw\-(n|u)|c55\/|capi|ccwa|cdm\-|cell|chtm|cldc|cmd\-|co(mp|nd)|craw|da(it|ll|ng)|dbte|dc\-s|devi|dica|dmob|do(c|p)o|ds(12|\-d)|el(49|ai)|em(l2|ul)|er(ic|k0)|esl8|ez([4-7]0|os|wa|ze)|fetc|fly(\-|_)|g1 u|g560|gene|gf\-5|g\-mo|go(\.w|od)|gr(ad|un)|haie|hcit|hd\-(m|p|t)|hei\-|hi(pt|ta)|hp( i|ip)|hs\-c|ht(c(\-| |_|a|g|p|s|t)|tp)|hu(aw|tc)|i\-(20|go|ma)|i230|iac( |\-|\/)|ibro|idea|ig01|ikom|im1k|inno|ipaq|iris|ja(t|v)a|jbro|jemu|jigs|kddi|keji|kgt( |\/)|klon|kpt |kwc\-|kyo(c|k)|le(no|xi)|lg( g|\/(k|l|u)|50|54|\-[a-w])|libw|lynx|m1\-w|m3ga|m50\/|ma(te|ui|xo)|mc(01|21|ca)|m\-cr|me(rc|ri)|mi(o8|oa|ts)|mmef|mo(01|02|bi|de|do|t(\-| |o|v)|zz)|mt(50|p1|v )|mwbp|mywa|n10[0-2]|n20[2-3]|n30(0|2)|n50(0|2|5)|n7(0(0|1)|10)|ne((c|m)\-|on|tf|wf|wg|wt)|nok(6|i)|nzph|o2im|op(ti|wv)|oran|owg1|p800|pan(a|d|t)|pdxg|pg(13|\-([1-8]|c))|phil|pire|pl(ay|uc)|pn\-2|po(ck|rt|se)|prox|psio|pt\-g|qa\-a|qc(07|12|21|32|60|\-[2-7]|i\-)|qtek|r380|r600|raks|rim9|ro(ve|zo)|s55\/|sa(ge|ma|mm|ms|ny|va)|sc(01|h\-|oo|p\-)|sdk\/|se(c(\-|0|1)|47|mc|nd|ri)|sgh\-|shar|sie(\-|m)|sk\-0|sl(45|id)|sm(al|ar|b3|it|t5)|so(ft|ny)|sp(01|h\-|v\-|v )|sy(01|mb)|t2(18|50)|t6(00|10|18)|ta(gt|lk)|tcl\-|tdg\-|tel(i|m)|tim\-|t\-mo|to(pl|sh)|ts(70|m\-|m3|m5)|tx\-9|up(\.b|g1|si)|utst|v400|v750|veri|vi(rg|te)|vk(40|5[0-3]|\-v)|vm40|voda|vulc|vx(52|53|60|61|70|80|81|83|85|98)|w3c(\-| )|webc|whit|wi(g |nc|nw)|wmlb|wonu|x700|yas\-|your|zeto|zte\-/i.test(a.substr(0,4)))check = true})(navigator.userAgent||navigator.vendor||window.opera);
	return check;
}

// ~~~~~~~ uisearch.js v1.0.0 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ //
;( function( window ) {
	
	'use strict';
	
	!window.addEventListener && window.Element && (function () {
	   function addToPrototype(name, method) {
		  Window.prototype[name] = HTMLDocument.prototype[name] = Element.prototype[name] = method;
	   }
	 
	   var registry = [];
	 
	   addToPrototype("addEventListener", function (type, listener) {
		  var target = this;
	 
		  registry.unshift({
			 __listener: function (event) {
				event.currentTarget = target;
				event.pageX = event.clientX + document.documentElement.scrollLeft;
				event.pageY = event.clientY + document.documentElement.scrollTop;
				event.preventDefault = function () { event.returnValue = false };
				event.relatedTarget = event.fromElement || null;
				event.stopPropagation = function () { event.cancelBubble = true };
				event.relatedTarget = event.fromElement || null;
				event.target = event.srcElement || target;
				event.timeStamp = +new Date;
	 
				listener.call(target, event);
			 },
			 listener: listener,
			 target: target,
			 type: type
		  });
	 
		  this.attachEvent("on" + type, registry[0].__listener);
	   });
	 
	   addToPrototype("removeEventListener", function (type, listener) {
		  for (var index = 0, length = registry.length; index < length; ++index) {
			 if (registry[index].target == this && registry[index].type == type && registry[index].listener == listener) {
				return this.detachEvent("on" + type, registry.splice(index, 1)[0].__listener);
			 }
		  }
	   });
	 
	   addToPrototype("dispatchEvent", function (eventObject) {
		  try {
			 return this.fireEvent("on" + eventObject.type, eventObject);
		  } catch (error) {
			 for (var index = 0, length = registry.length; index < length; ++index) {
				if (registry[index].target == this && registry[index].type == eventObject.type) {
				   registry[index].call(this, eventObject);
				}
			 }
		  }
	   });
	})();
	
	!String.prototype.trim && (String.prototype.trim = function() {
		return this.replace(/^\s+|\s+$/g, '');
	});

	function UISearch( el, options ) {	
		this.el = el;
		this.inputEl = el.querySelector( 'form > input.AnitaBakai-search-input' );
		this._initEvents();
	}

	UISearch.prototype = {
		_initEvents : function() {
			var self = this,
				initSearchFn = function( ev ) {
					ev.stopPropagation();
					// trim its value
					self.inputEl.value = self.inputEl.value.trim();
					
					if( !classie.has( self.el, 'AnitaBakai-search-open' ) ) { // open it
						ev.preventDefault();
						self.open();
					}
					else if( classie.has( self.el, 'AnitaBakai-search-open' ) && /^\s*$/.test( self.inputEl.value ) ) { // close it
						ev.preventDefault();
						self.close();
					}
				}

			this.el.addEventListener( 'click', initSearchFn );
			this.el.addEventListener( 'touchstart', initSearchFn );
			this.inputEl.addEventListener( 'click', function( ev ) { ev.stopPropagation(); });
			this.inputEl.addEventListener( 'touchstart', function( ev ) { ev.stopPropagation(); } );
		},
		open : function() {
			var self = this;
			classie.add( this.el, 'AnitaBakai-search-open' );
			// focus the input
			if( !mobilecheck() ) {
				this.inputEl.focus();
			}
			// close the search input if body is clicked
			var bodyFn = function( ev ) {
				self.close();
				this.removeEventListener( 'click', bodyFn );
				this.removeEventListener( 'touchstart', bodyFn );
			};
			document.addEventListener( 'click', bodyFn );
			document.addEventListener( 'touchstart', bodyFn );
		},
		close : function() {
			this.inputEl.blur();
			classie.remove( this.el, 'AnitaBakai-search-open' );
		}
	}

	window.UISearch = UISearch;
} )( window );



// ~~~~~~~ FORM ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ //
var theForm = document.getElementById( 'theForm' );

new stepsForm( theForm, {
	onSubmit : function( form ) {
		// hide form
		classie.addClass( theForm.querySelector( '.simform-inner' ), 'hide' );

		jQuery('#theForm').ajaxForm(function() {}); 
		jQuery('[name="submit_contact"]').click();

		// let's just simulate something...
		var messageEl = theForm.querySelector( '.final-message' );
		messageEl.innerHTML = 'Thank you! We\'ll be in touch.';
		classie.addClass( messageEl, 'show' );
	}
} );

gallery2/js/cj-galleryjack-settings.js
/* --------------------------------------------- */
/* Author: http://codecanyon.net/user/CodingJack */
/* --------------------------------------------- */


// ********************************
// ENTER YOUR CUSTOM SETTINGS BELOW
// ********************************

function galleryJackSettings() {
			
	/* The Slideshow Settings */
	var slideshowSettings = {
			
		/* Choose to autoPlay the gallery, true/false */
		autoPlay: true,
					
		/* Choose to randomize the image order for each category, true/false */
		randomizeImages: false,
					
		/* The transition delay between images */
		transitionDelay: 4000,
		
		/* The amount of time the transition takes to complete */
		transitionTime: 500,
		
		/* Choose to hide the thumbnails when the gallery first loads */
		hideThumbnails: false,
		
		/* Choose to use a tooltip for thumbnails, true/false */
		useThumbToolTip: true,
		
		/* The thumbnail circle that animates between images (if autoPlay is turned on) */
		useCanvasAnimation: true,
					
		/* Enter 0 to always display maximum number of thumbs based on monitor/gallery size */
		maxThumbs: 0,
					
		/* The opacity level for the thumbnail hover effect, 0-1 */
		thumbHoverOpacity: 0.5,
		
		/* The opacity level for category text mouseover */
		categoryOpacity: 0.65,
		
		/* Left/Right Arrows will control slideshow, true/false */
		keyboardControls: true,
		
		/* Disable right click to help protect your images, true/false */
		disableRightClick: true,
		
		/* Choose to help Google index the image deep-links, true/false */
		seoCrawlable: false,
		
		/* If "fullScreen" is set to "false", choose a background color (applicable for "fit" scaling mode */
		backgroundColor: "#000",
		
		/* If using the fullscreen version, this number will act as an offset for the width */
		offsetWidth: 0,
		
		/* If using the fullscreen version, this number will act as an offset for the height */
		offsetHeight: 0,
		
		/* If using the fullscreen version, this number will act as an offset left position */
		offsetX: 0,
		
		/* If using the fullscreen version, this number will act as an offset for the top position */
		offsetY: 0,
		
		
		/* ******************************************************************** */
		/* The values below are set in the JavaScript portion of the html files */
		/* ******************************************************************** */
		
		/* Choose to show the image info when the gallery loads, true/false */
		showInfoByDefault: cjShowInfoByDefault,
		bgClickLoadsNext: cjBgClickLoadsNext,
		fullScreen: cjFullScreenMode,
		customHeight: cjCustomHeight,
		useDeepLink: cjDeepLinkMode,
		customWidth: cjCustomWidth,
		useThumbs: cjUseThumbs
				
	},
	
	/* The Video Settings */
	videoSettings = {
	
		/* The video width if video is used */
		videoWidth: 640,
		
		/* The video height if video is used */
		videoHeight: 360,
		
		/* The default volume for video */
		videoVolume: 75,
		
		/* Choose to autoplay video, true/false */
		autoPlayVideo: true,
		
		/* The video Flash fallback swf url */
		videoFallbackSwf: "flash/gallery_video.swf"
		
	},
	
	/* The Music Settings */
	musicSettings = {
				
		/* Music Default Volume, this should be in multiples of 5 or 10 */
		volume: 75,
					
		/* Choose to AutoPlay Slideshow */
		autoPlay: cjAutoPlayMusic,
					
		/* Choose to randomize the song list, true/false */
		randomizeSongs: false,
					
		/* The list of songs to play */
		songList: [
						
			/* title: The title of your song (delete param for no title), url: the url to the mp3 version of the song */
            {title: "Bluegestalt - Becoming", url: "music/song4.mp3"},
			//{title: "DanielKeller - Rock the Show", url: "music/song3.mp3"},
			{title: "Leto - Follow Me", url: "music/song2.mp3"},
			
			/* Last item doesn't have a comma at the end */
			//{title: "Caseilla - U Let Me Down Again", url: "music/song1.mp3"}
			
		]

	},
			
	/* Contact Form Settings */
	contactSettings = {
				
		/* The generic field error message */
		emptyTextMessage: "Please fill in your", 
				
		/* The email error message */
		emailErrorMessage: "Please enter a valid email",
				
		/* The form submit success message */
		successMessage: "Your message has been sent",
				
		/* The contact form's php file */
		contactPhpFile: "php/cj-contact.php" 
				
	},
			
	/* Social Settings */
	socialSettings = {
				
		/* Facebook Fanpage */
		facebookUser: "50andJACK",
				
		/* Twitter Username */
		twitterUser: "CodingJack",
		
		/* Google Plus Profile Link */
		googlePlusId: "116330055061518644270",
		
		/* ******************************************************************* */
		/* The social icons below are optional and can be reordered or deleted */
		/* ******************************************************************* */
		
		/* Set this to an empty string to not use */
		facebookLike: "<div id='cj-facebook'><iframe src='http://www.facebook.com/plugins/like.php?locale=en_US&href=http://www.codingjack.com&layout=button_count&show_faces=false&width=500&action=like&font&colorscheme=light&height=23' scrolling='no' frameborder='0' style='border:none; overflow:hidden; width:85px; height:23px;' allowTransparency='true'></iframe></div>",
				
		/* Set this to an empty string to not use */
		tweetThis: "<div id='cj-twitter'><a href='http://twitter.com/share' class='twitter-share-button' data-count='horizontal'>Tweet</a><script async defer type='text/javascript' src='http://platform.twitter.com/widgets.js'></scr" + "ipt></div>",
		
		/* Set this to an empty string to not use */
		googlePlus: "<div id='cj-google-plus'><div class='g-plusone' data-size='medium' data-href='" + document.URL + "'></div></div><script async defer type='text/javascript'>(function() {var po = document.createElement('script'); po.type = 'text/javascript'; po.async = true; po.src = 'https://apis.google.com/js/plusone.js'; var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(po, s);})();</scr" + "ipt>",
		
		/* Set this to an empty string to not use */
		twitterIcon: "<div id='cj-twitter-icon'><a href='http://twitter.com/{twitterUser}' target='_blank'><img src='img/cj_gallery/graphics/twitter.png' width='20' height='20' /></a></div>",
				
		/* Set this to an empty string to not use */
		facebookIcon: "<div id='cj-facebook-icon'><a href='http://www.facebook.com/{facebookUser}' target='_blank'><img src='img/cj_gallery/graphics/facebook.png' width='20' height='20' /></a></div>",
		
		/* Set this to an empty string to not use */
		googlePlusIcon: "<div id='cj-google-icon'><a rel='author' href='https://plus.google.com/{googlePlusId}' target='_blank'><img src='http://www.google.com/images/icons/ui/gprofile_button-32.png' width='20' height='20'></a></div>"
		
		
		/* You can add any icon or html to this section.  Just give it a unique name like the example below */
		
		/*
		
		, genericIconOne: "icon html here"
		
		*/
		
		
	},
	
	
	/* If cjAjaxVersion = true (declared in the html file), enter your ajax version settings here */
	ajaxSettings = {
		
		/* The path to your large images */
		imagePath: "img/cj_gallery/photos/",
		
		/* The thumbnail folder name (must be located directly inside the "imagePath" folder */
		thumbFolder: "thumbs",
		
		/* The gallery stretch mode and transition type (view the help documentation for options) */
		className: "cj-stretch cj-fade",
		
		/* The width and height of your thumbnails */
        thumbSize: "100x67",
		
		/* Choose to use the image names for the thumbnail tool-tips, true/false */
		imageNameTips: true,
		
		/* The ajax version's php url */
		ajaxPhpUrl: "php/cj-ajax.php"
		
	};
	
	
	
	// **************************************************
	// **************************************************
	// DEVELOPER SETTINGS, NO NEED TO EDIT ANYTHING BELOW
	// **************************************************
	// **************************************************
	
	// Setup jQuery ready function
	if(!cjAjaxVersion) {
		
		jQuery(document).ready(function() {
			
			startIt();
			
		});
		
	}
	else {
		
		jQuery(document).ready(function() {
                
			jQuery.address.init(addressReady);
                
        });
		
	}
	
	// Make the ajax call
	function addressReady() {
		
		jQuery.getJSON(ajaxSettings.ajaxPhpUrl + "?path=../" + ajaxSettings.imagePath, ajaxLoaded);
				
	}
	
	// Recieve the folder and image url data from php
	function ajaxLoaded(data, response) {
		
    	if(response.toLowerCase() === "success") {
			   
			var thumbFolder = ajaxSettings.thumbFolder,
			nameTips = ajaxSettings.imageNameTips,
			className = ajaxSettings.className,
			thumbSize = ajaxSettings.thumbSize,
			bgDiv = jQuery("#cj-bg-images"),
			images = data[1],
			names = data[0],
			leg = names.length,
			imageList,
			shaved,	
			large,
			thumb,
			jLeg,
			cat,
			img,
			li,
			ul,
            j;    
			
			if(bgDiv.length) {
			
				for(var i = 0; i < leg; i++) {
							
					cat = jQuery("<ul />").attr("title", names[i]).attr("class", className).attr("data-rel", thumbSize);
							
					imageList = images[i];
					jLeg = imageList.length;
					   
					for(j = 0; j < jLeg; j++) {
								
						li = jQuery("<li />").appendTo(cat);
						ul = jQuery("<ul />").appendTo(li);
								
						img = imageList[j].split("../").join("");
						shaved = img.substr(img.lastIndexOf("/"), img.length - 1);
						
						large = jQuery("<li />").attr("class", "cj-large").attr("title", img).appendTo(ul);
						thumb = jQuery("<li />").attr("class", "cj-thumb").attr("title", img.split(shaved).join("/" + thumbFolder + shaved)).appendTo(ul);
						
						if(nameTips) thumb.text(shaved.split(".jpg").join("").split("/").join(""));
								
					}
							
					bgDiv.append(cat);
							
				}
						
				startIt();
				
			}
                    
		}
		else {
                
			alert("Ajax Response: " + response);
                    
		}
                    
	}
	
	// Activate the GalleryJack scripts
	function startIt() {
			
		jQuery.cjBgSlideshow(slideshowSettings, videoSettings);
		
		if(typeof jQuery.address !== "undefined") jQuery.address.crawlable(slideshowSettings.seoCrawlable);
		if(typeof jQuery.cjTopRightButtons !== "undefined") jQuery.cjTopRightButtons(contactSettings, socialSettings);
		if(typeof jQuery.cjMusicPlayer !== "undefined") jQuery.cjMusicPlayer(musicSettings);
		if(typeof jQuery.cjVideoPlayer !== "undefined") jQuery.cjVideoPlayer(videoSettings);
				
	}
	
}

// kick it off
galleryJackSettings();

